huaweicloud/resource_huaweicloud_elb_loadbalancer.go

```diff
@@ -0,0 +1,478 @@
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceLoadBalancerV3() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLoadBalancerV3Create,
+		Read:   resourceLoadBalancerV3Read,
+		Update: resourceLoadBalancerV3Update,
+		Delete: resourceLoadBalancerV3Delete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"availability_zone": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MinItems: 1,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"cross_vpc_backend": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"ipv4_subnet_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"ipv6_network_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"ipv6_bandwidth_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"ipv4_address": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"ipv4_eip_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				ConflictsWith: []string{
+					"iptype", "bandwidth_charge_mode", "bandwidth_size", "sharetype",
+				},
+			},
+
+			"iptype": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				RequiredWith: []string{
+					"bandwidth_charge_mode", "bandwidth_size", "sharetype",
+				},
+				ConflictsWith: []string{"ipv4_eip_id"},
+			},
+
+			"bandwidth_charge_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				RequiredWith: []string{
+					"iptype", "bandwidth_size", "sharetype",
+				},
+				ConflictsWith: []string{"ipv4_eip_id"},
+			},
+
+			"sharetype": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				RequiredWith: []string{
+					"iptype", "bandwidth_charge_mode", "bandwidth_size",
+				},
+				ConflictsWith: []string{"ipv4_eip_id"},
+			},
+
+			"bandwidth_size": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				RequiredWith: []string{
+					"iptype", "bandwidth_charge_mode", "sharetype",
+				},
+				ConflictsWith: []string{"ipv4_eip_id"},
+			},
+
+			"l4_flavor_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"l7_flavor_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"name": {
```

Comment:
this parameter is **required** in the schame while it is optional in docs.

Reply:
will update in the following up patch set.